File: code/js/src/Dto/output/messages/MessagesPaginatedOutputModel.ts
import { MessageOutputModel } from './MessageOutputModel';
import { PaginationOutputModel } from '../PaginationOutputModel';

/**
 * Output model for messages paginated
 *
 * @param messages The messages.
 * @param pagination The pagination information.
 */
export interface MessagesPaginatedOutputModel {
    messages: MessageOutputModel[];
    pagination: PaginationOutputModel;
}

File: code/server/jvm/im-http-api/src/main/kotlin/im/api/model/output/messages/MessagesPaginatedOutputModel.kt
package im.api.model.output.messages

import im.api.model.output.PaginationOutputModel
import im.domain.messages.Message
import im.repository.pagination.PaginationInfo

data class MessagesPaginatedOutputModel(
    val messages: List<MessageOutputModel>,
    val pagination: PaginationOutputModel,
) {
    companion object {
        fun fromMessages(
            messages: List<Message>,
            pagination: PaginationInfo,
        ): MessagesPaginatedOutputModel =
            MessagesPaginatedOutputModel(
                messages.map { MessageOutputModel.fromDomain(it) },
                PaginationOutputModel.fromPagination(pagination),
            )
    }
}
